Feat: Add documentation for servicediscovery filters (#1142)
localstack/docs: content/en/user-guide/aws/servicediscovery/index.md

The output will consist of the resource ID, and you can further use the [`DiscoverInstances`](https://docs.aws.amazon.com/cloud-map/latest/api/API_DiscoverInstances.html) API. This API allows you to query the DNS records associated with the service and perform various operations.

To explore the DNS records of your service and perform other operations, refer to the [AWS CLI documentation](https://docs.aws.amazon.com/cli/latest/reference/servicediscovery/index.html) for comprehensive instructions and examples.

### Using filters
Filters can be used to narrow down the results of a list operation. Filters are supported for the following operations:

- list-namespaces
- list-services

> - discover-instances

Author: interestingly the AWS docs for `list-services`are missing the `--filter`: https://docs.aws.amazon.com/cli/latest/reference/ecs/list-services.html

> - discover-instances

Author: added in: https://github.com/localstack/docs/pull/1142/commits/bc9002d7e36b80c4059c40d9d4d349095467ce1f

> Using `list-namespaces` you can filter for the parameters `TYPE`, `NAME`, `HTTP_NAME`. Using `list-services` it is only possible to filter for `NAMESPACE_ID`.
> Both `list-services` and `list-namespaces` support `EQ` (default condition if not specified) and `BEGINS_WITH` as conditions. Both conditions and only support a single value to match by. The following examples demonstrate how to use filters with these operations:

Reviewer: ```suggestion
Using `list-namespaces` you can filter for the parameters `TYPE`, `NAME`, `HTTP_NAME`.
Using `list-services` it is only possible to filter for `NAMESPACE_ID`.
Both `list-services` and `list-namespaces` support `EQ` (default condition if not specified) and `BEGINS_WITH` as conditions.
Both conditions and only support a single value to match by.
The following examples demonstrate how to use filters with these operations:
```

I believe that the standard is to keep one sentence in one line for easier readability. I was unable to find where in notion where it's mentioned. :(

Author: fixed for complete do in: https://github.com/localstack/docs/pull/1142/commits/2966f96624d439a148cd69c97c528511b9ce262b

{{< command >}}

> awslocal servicediscovery list-namespaces --filters "Name=HTTP_NAME,Values=['example-namespace'],Condition=EQ"

Author: fixed in: https://github.com/localstack/docs/pull/1142/commits/4bba24d03f5e98a5a0863f419cc505bea61fb070

> awslocal servicediscovery list-services --filters "Name=NAMESPACE_ID,Values=['id_to_match']"

Author: fixed in: https://github.com/localstack/docs/pull/1142/commits/4bba24d03f5e98a5a0863f419cc505bea61fb070

{{< /command >}}

> The command `discover-instance` supports parameters and optional parameters as filter criteria. Conditions in parameters must match return values, while if one ore more conditions in optional parameters match, the subset is returned, if no conditions in optional parameters match, all unfiltered results are returned.

Reviewer: ```suggestion
The command `discover-instance` supports parameters and optional parameters as filter criteria.
Conditions in parameters must match return values.
When one or more conditions in the optional parameters are met, a subset is returned; however, if no conditions are met, all unfiltered results are returned.
```

Author: fixed for complete do in: https://github.com/localstack/docs/pull/1142/commits/2966f96624d439a148cd69c97c528511b9ce262b

This command will only return instances where the parameter `env` is equal to `fuu`:
{{< command >}}

> aws servicediscovery discover-instances --namespace-name example-namespace --service-name example-service --query-parameters "env"="fuu"

Author: fixed in: https://github.com/localstack/docs/pull/1142/commits/4bba24d03f5e98a5a0863f419cc505bea61fb070

{{< /command >}}

This command instead will return all instances where the optional parameter `env` is equal to `bar`, but if no instances match, all instances are returned:
{{< command >}}

> aws servicediscovery discover-instances --namespace-name example-namespace --service-name example-service --optional-parameters "env"="bar"

Author: fixed in: https://github.com/localstack/docs/pull/1142/commits/4bba24d03f5e98a5a0863f419cc505bea61fb070
